### DISCUSSION TITLE
Is it possible to dynamically add websites on the self-hosted version?

### DISCUSSION THREAD

**Discussion by ozziestraw** [plausible/analytics#805] [Self-Hosted Support]
I'm planning on using Plausible, but need the ability to dynamically add websites to be tracked.

Is this possible?

Also, is there a way to retrieve data from the any of those dynamically added websites (e.g. like the API, but for the self-hosted version).

Plus, would it make sense to track thousands of websites from a single instance? I assume it just depends on how powerful the server is right?

Thanks!

**Comment by ukutaht** [comment]
Sorry it's not possible at the moment. We might have something like this available later this year.

The number of sites is less important than the number of pageviews. We run a fairly large instance ourselves, ingesting about 150m pageviews per month so that's definitely possible :)

**Reply by ozziestraw** [reply]
Thanks for the reply.

Would that only be for the Cloud/Managed version of Plausible? Or would it work for the open-sourced version too

I'm currently using the public version of the API via https://example.com/api/stats/ - is there a way to get that to work for sites that aren't made public (visibility settings)?

**Reply by metmarkosaric** [reply]
Both versions of Plausible are fully open source. There's no proprietary version of Plausible.

Our stats API is available to all of the websites, it doesn't matter if your stats are public or not. You just need to create the API key in your account settings. You can read more here: https://plausible.io/docs/stats-api

**Reply by ukutaht** [reply]
> Would that only be for the Cloud/Managed version of Plausible? Or would it work for the open-sourced version too

It does make it much easier for other hosting companies to compete with our SaaS offering. So we are thinking about ways to protect against that. I can't promise that it will be included in the self-hosted version.

> is there a way to get that to work for sites that aren't made public (visibility settings)?

Don't think so, those endpoints require an authenticated session to return the stats. Since our cookie is only valid on the same domain as the app, it makes it impossible to use the internal API on another domain with a private site.
